packages/devtools_app/test/flutter/event_details_test.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/profiler/flutter/cpu_profiler.dart';
+import 'package:devtools_app/src/timeline/flutter/event_details.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'wrappers.dart';
+
+void main() {
+  group('EventDetails', () {
+    EventDetails eventDetails;
+    testWidgets('builds for UI event', (WidgetTester tester) async {
+      eventDetails = EventDetails(stubUiEvent);
+      await tester.pumpWidget(wrap(eventDetails));
+      expect(find.byType(EventDetails), findsOneWidget);
+      expect(find.byType(CpuProfilerView), findsOneWidget);
+      expect(find.byType(EventSummary), findsNothing);
+    });
+
+    testWidgets('builds for GPU event', (WidgetTester tester) async {
+      eventDetails = EventDetails(stubGpuEvent);
+      await tester.pumpWidget(wrap(eventDetails));
+      expect(find.byType(EventDetails), findsOneWidget);
+      expect(find.byType(CpuProfilerView), findsNothing);
+      expect(find.byType(EventSummary), findsOneWidget);
+    });
+
+    testWidgets('builds for ASYNC event', (WidgetTester tester) async {
+      eventDetails = EventDetails(stubAsyncEvent);
+      await tester.pumpWidget(wrap(eventDetails));
+      expect(find.byType(EventDetails), findsOneWidget);
+      expect(find.byType(CpuProfilerView), findsNothing);
+      expect(find.byType(EventSummary), findsOneWidget);
+    });
+  });
+
+  group('EventSummary', () {
```

Comment:
add a todo to make at least one of tests check that the data returned actually looks right. e.g. use a golden image test or check that the right text is generated for the fields.

Reply:
done.